Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -171,11 +160,47 @@ void UPSWorldSubsystem::OnWorldBeginPlay(UWorld& InWorld)
 void UPSWorldSubsystem::Deinitialize()
 {
 	Super::Deinitialize();
-	PSHUDComponentInternal = nullptr;
-	PSCurrentSpotComponentInternal = nullptr;
-	SaveGameDataInternal = nullptr;
-	ProgressionSettingsDataInternal.Empty();
-	StarDynamicProgressMaterial = nullptr;
+}
+
+// Invoked after a game feature plugin is unloaded
+void UPSWorldSubsystem::OnGameFeatureUnloading(const UGameFeatureData* GameFeatureData, const FString& PluginURL)
+{
+	DestroyStarActors();
+	IGameFeatureStateChangeObserver::OnGameFeatureUnloading(GameFeatureData, PluginURL);
+}
+
+// Invoked in the early stages of the game feature plugin loading phase
+void UPSWorldSubsystem::OnGameFeatureLoading(const UGameFeatureData* GameFeatureData, const FString& PluginURL)
+{
+	const FString ProgressionPlugin = TEXT("/ProgressionSystem/ProgressionSystem");
+	if (PluginURL.Contains(ProgressionPlugin))
+	{
+		WolrdSubSystemInitialize();
+	}
+
+	IGameFeatureStateChangeObserver::OnGameFeatureLoading(GameFeatureData, PluginURL);
+}
+
+// Is called to initialize the world subsystem
+void UPSWorldSubsystem::WolrdSubSystemInitialize()
+{
+	UGameFeaturesSubsystem::Get().AddObserver(this);
+
+	UPSDataAsset::Get();
```

Comment:
`UPSDataAsset::Get();` looks like a hack. Please remove it. Any call of `UPSDataAsset::Get();` will load data asset on demand

Reply:
removed https://github.com/h4rdmol/ProgressionSystem/pull/26/commits/b09e88a264e6b09844f573d09e9f6726b084f2ac